### PR TITLE
test(e2e): dapp list display

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Moved some functions shared between `DappListRecent.js` `DappListDisplay.js` to `DappList.spec.js`.
Added several tests for the display of the dapp list.

```
  Dapp List
    Dapp List Display
      ✓ should show correct featured dapp (4306 ms)
      ✓ should show dapp info icon and subsequent modal (3088 ms)
      ✓ should show dapp bottom sheet when dapp is selected (1405 ms)
      ✓ should open internal webview with correct dapp when dapp opened (2238 ms)
      ✓ should correctly filter dapp list based on user agent :ios: (23106 ms)
```
### Test plan

Tested locally on iOS.
Tested on Android in CI.

### Related issues

- Fixes RET-415

### Backwards compatibility

Yes

### CODE REVIEW

File: e2e/src/usecases/DappListDisplay.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    await waitFor(element(by.text(`Go to ${dappList.applications[0].name}`)))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
+  })
+
+  it('should open internal webview with correct dapp when dapp opened', async () => {
+    await element(by.id('ConfirmDappButton')).tap()
+    await waitFor(element(by.id(`WebViewScreen/${dappList.applications[0].name}`)))
+      .toBeVisible()
+      .withTimeout(10 * 1000)
```
Comment (kathaypacific) [inline e2e/src/usecases/DappListDisplay.js:41]: feel free to ignore this suggestion, but should we also add an assertion to test the domain of the dapp url should be on the header of the screen?

File: e2e/src/usecases/DappListDisplay.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  })
+
+  it('should correctly filter dapp list based on user agent :ios:', async () => {
+    const iOSDappList = await fetchDappList('celo/')
+    await reloadReactNative()
+    await navigateToDappList()
+    // Give a few seconds for the dapp list to load
+    await sleep(5 * 1000)
+    const dappCards = await getElementTextList('DappCard')
+    jestExpect(dappCards.length).toEqual(iOSDappList.applications.length)
```
Comment (kathaypacific) [inline e2e/src/usecases/DappListDisplay.js:51]: nice!

File: src/dappsExplorer/DAppsExplorerScreen.tsx [hunk 3]
```diff
@@ -97,7 +97,7 @@ export function DAppsExplorerScreen() {
     <SafeAreaView style={styles.safeAreaContainer} edges={['top']}>
       <DrawerTopBar
         middleElement={<Text style={headerStyles.headerTitle}>{t('dappsScreen.title')}</Text>}
-        rightElement={<TopBarIconButton icon={<Help />} onPress={onPressHelp} />}
+        rightElement={<TopBarIconButton testID="HelpIcon" icon={<Help />} onPress={onPressHelp} />}
```
Comment (kathaypacific) [inline src/dappsExplorer/DAppsExplorerScreen.tsx:100]: nit: DAppsExplorerScreen/HelpIcon

File: src/dappsExplorer/FeaturedDappCard.tsx [hunk 4]
```diff
@@ -20,7 +20,9 @@ function FeaturedDappCard({ dapp, onPressDapp }: Props) {
       <Touchable style={styles.pressableCard} onPress={onPress} testID="FeaturedDapp">
         <>
           <View style={styles.itemTextContainer}>
-            <Text style={styles.title}>{dapp.name}</Text>
+            <Text testID="FeaturedDappCard/Name" style={styles.title}>
```
Comment (kathaypacific) [inline src/dappsExplorer/FeaturedDappCard.tsx:23]: nit: FeaturedDapp to match the testID above

**Review by kathaypacific — COMMENTED** [review]
☄️

File: e2e/src/usecases/DappListDisplay.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
+import { reloadReactNative } from '../utils/retries'
+import { waitForElementId, getElementText, getElementTextList, sleep } from '../utils/utils'
+import { navigateToDappList, fetchDappList } from '../DappList.spec'
```
[the remark this answers is not in the document]
Comment (MuckT) [reply e2e/src/usecases/DappListDisplay.js:3]: Fixed in bf1fa9690c3039a4dbe498603a6f04ff7023e473!

File: src/dappsExplorer/DAppsExplorerScreen.tsx (hunk 3 above)
Comment (MuckT) [reply src/dappsExplorer/DAppsExplorerScreen.tsx]: Fixed in afaa22c75524205d035e80d06bc31f3fef80a8ff!

File: src/dappsExplorer/FeaturedDappCard.tsx (hunk 4 above)
Comment (MuckT) [reply src/dappsExplorer/FeaturedDappCard.tsx]: Fixed in afaa22c75524205d035e80d06bc31f3fef80a8ff!

File: e2e/src/DappList.spec.js [hunk 6]
```diff
@@ -1,6 +1,73 @@
...
+  } catch {}
+  return dappList
+}
+
+/**
+ * From the home screen, navigate to the dapp explorer screen
+ * Disable detox device synchronization to allow for dapp list to load
+ * https://github.com/wix/Detox/issues/2799
+ */
+export async function navigateToDappList() {
```
Comment (kathaypacific) [inline e2e/src/DappList.spec.js:35]: these functions are in the dappList utils file now right? they can be removed here?

File: e2e/src/usecases/DappListDisplay.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (kathaypacific) [reply e2e/src/usecases/DappListDisplay.js:3]: i think that this was fixed in the recently used dapps test, but still remains in this file? at least, i still see these changes in the latest commit of this branch https://github.com/valora-inc/wallet/pull/3321/commits/20c345855724e2b062d9b8699851f154b5868e1c

**Review by kathaypacific — APPROVED** [review]
the DappListDisplay usecase should use functions from the utils/DappList file rather than from the spec file, but other than that LGTM! ✨

**Comment by MuckT** [comment]
@kathaypacific imports should all be fixed in bd14cee1870047a9666a59f22d7a9dea7719e7b4!

**Pull request merged**